### ISSUE TITLE
Cmake View : show only files not folder in the tree view

### ISSUE THREAD

**Issue by mickae1** [microsoft/vscode-cmake-tools#3021]
### Brief Issue Summary

When I click on the Cmake View ( the CMake icon on the left)

I can see my projects, but under my project, i see the folder and after the file, example : 

MyProject
--Folder1
-----File1.c
--Folder2
-----File2.c

I prefet to have a view like that : 
MyProject
-----File1.c
-----File2.c

I'm saying that, because in my works, I have a lot of files, and a lot of different folder.

Thank you,

**Comment by bobbrow** [comment]
Thank you for the feature idea.  We may not be able to get to this immediately, but would accept a PR from someone in the community if someone can get to it before us.

**Comment by mickae1** [comment]
@bobbrow can you tell me which files I should look into ?

thank you,

**Comment by bobbrow** [comment]
@mickae1 `src/projectOutlineProvider.ts` is where the outline code lives.  The information for the outline view itself is declared in `package.json`.  We may want to add a toggle button or context menu item in the view that lets you change the view type.

**Comment by mickae1** [comment]
Ok, I don't have the level to modify the source code, unfortunately.
I hope that someone can make it !

**Comment by bobbrow** [comment]
Correct, you cannot add branches directly to this repository.  Have you used the "fork" feature in GitHub yet?  If you click that button, GitHub will create a "fork" of the repository where you can commit to the main branch.  Then after you push your changes to the server, you can create a "pull request" back into the main repository (this one).

**Comment by mickae1** [comment]
I successfully make what I wanted with this simple modification :)

instead of this line :
https://github.com/microsoft/vscode-cmake-tools/blob/main/src/projectOutlineProvider.ts#L366
 ``` 
const relpath = path.relative(this.sourceDir, src_dir);
```

I put this:

 ```
               let relpath = path.relative(this.sourceDir, src_dir);
                switch(src.split('.').pop())
                {
                    case 'c':
                    case 'cpp':
                        relpath = 'source';
                        break;
                    case 'h':
                    case 'hpp':
                        relpath = 'header';
                        break;
                }
```

but now, I've to made it optionally with a button, or something else. Maybe in a preference button ?

I'm not an expert in your project.
I don't know, where to start.

**Comment by benmcmorran** [comment]
Have you opened a pull request before? That's typically how code changes are proposed for open source GitHub projects. [This article](https://jarv.is/notes/how-to-pull-request-fork-github/) has a good summary of how the process works.
